tests/unit/helpers.js

```diff
@@ -1478,6 +1478,7 @@ export const createExperimentData = ({
   return { [id]: variantId };
 };
 
+/* eslint-disable testing-library/no-node-access */
```

Comment:
Disabling this rule here for the library functions while keeping it enabled globally feels a little like a fake seal of approval to me.

Reply:
Can you explain a bit more what you mean/suggest? I didn't disable this rule globally to discourage us from accessing nodes directly from within individual tests. Are you saying that you think we should just disable this globally, rather than "pretend" we are actually respecting the rule, or that we shouldn't be writing any code that requires us to disable this anywhere?

It's true that accessing elements based on their class names and tag names is frowned upon by the framework. We could avoid both of these things by using [`ByTestId`](https://testing-library.com/docs/queries/bytestid), which is what they recommend, but I was loath to add a bunch of attributes into our HTML just for testing purposes. 

Personally, I don't think that the class names, nor the tag names, that we use in our tests are likely to change, much the same way a `data-testid` would not change, so I'm not sure I agree that there is an advantage to going the `ByTestId` route, but if the rest of the team believes that we would be better off getting rid of these custom queries (`ByClassName` and `ByTagName`), and using `ByTestId` instead, I am willing to make that change.

@eviljeff @diox WDYT?

Reply:
> I didn't disable this rule globally to discourage us from accessing nodes directly from within individual tests. Are you saying that you think we should just disable this globally, rather than "pretend" we are actually respecting the rule, or that we shouldn't be writing any code that requires us to disable this anywhere?

well, yes, in an ideal world.  And that having the helper functions encourages us to write tests in a non-standard way.

> It's true that accessing elements based on their class names and tag names is frowned upon by the framework. We could avoid both of these things by using [`ByTestId`](https://testing-library.com/docs/queries/bytestid), which is what they recommend, but I was loath to add a bunch of attributes into our HTML just for testing purposes.

test ids are worse imo - having to add extra markup, _just_ for tests, which you then have to scrub before bundling for distribution, seems the opposite of the principles of the framework.  I can only assume it's been made so terrible so you try your hardest to use ByRole, ByText, etc, instead.

so, what we're doing is probably the best solution - only having a couple of ways/places we disable the rule - because having random access to DOM nodes elsewhere is bad, and the alternative of test ids is worse.